### ISSUE TITLE
Contract Simplification + Source of Truth

### ISSUE THREAD

**Issue by mpeyfuss** [6529-Collections/nftdelegation#51]
Overall, I think there is a need to simplify a lot of the interface in order to get more adoption outside of the 6529 ecosystem. There are too many retrieve functions with lacking documentation, leading to expectations that aren't met in code. 

You mainly need just one or two functions to get if a delegated address is delegated or not for a specific use case, collection (or all collections), token, and at a specific timestamp. It would remove a lot of need to loop over delegate addresses found in many of the write functions. If you still want enumerations of all delegates or delegations, you can still achieve a simpler read interface than currently exists. The main point is to have one/two source of truth functions, plus a few other functions that provide nice enumerations for indexing.

**Comment by mpeyfuss** [comment]
Of course, I don't have the full vision that you all had when building it so some of the functions may be needed for reasons that I'm not thinking of, but I have tried to glean as much background info from the documentation as I can. Cheers!

**Comment by a2rocket** [comment]
The smart contract was designed with advanced features in mind that may be used or not in the future. In terms of documentation you can have a look at the draft for now docs located at https://docs.nftdelegation.com/ . There is also a Developer Center with sample integration examples.

**Comment by mpeyfuss** [comment]
Understood. There is a world where there are simple calls that can be made for easy integration, while keeping advanced features. Overall, there are some functions that seem to be geared towards simple integration, such as `retrieveTokenStatus` or `retrieveGlobalStatusOfDelegation`, but they have several pitfalls that could supply misleading information as expiration times aren't checked, `ALL_COLLECTIONS` aren't checked, etc. Overall, I think this is the biggest struggle point for me right now. There are seemingly simple check functions that I would love to use and are all I need, but they don't actually work as I would expect and to integrate, I would be forced to essentially just write a bunch of logic myself to loop through active delegations.
